src/server/scripts/Northrend/AzjolNerub/AzjolNerub/boss_krikthir_the_gatewatcher.cpp

```diff
@@ -630,15 +793,204 @@ class npc_watcher_silthik : public CreatureScript
 
                 DoMeleeAttackIfReady();
             }
+        };
 
-            private:
-                EventMap _events;
-                InstanceScript* _instance;
+        CreatureAI* GetAI(Creature* creature) const override
+        {
+            return GetInstanceAI<npc_anub_ar_shadowcasterAI>(creature);
+        }
+};
+
+class npc_skittering_swarmer : public CreatureScript
+{
+    public:
+        npc_skittering_swarmer() : CreatureScript("npc_skittering_swarmer") { }
+
+        struct npc_skittering_swarmerAI : public ScriptedAI
+        {
+            npc_skittering_swarmerAI(Creature* creature) : ScriptedAI(creature) { }
+
+            void InitializeAI() override
+            {
+                ScriptedAI::InitializeAI();
+                DoZoneInCombat();
+                if (Creature* gatewatcher = me->GetInstanceScript()->GetCreature(DATA_KRIKTHIR_THE_GATEWATCHER))
+                    gatewatcher->AI()->SetGUID(me->GetGUID());
```

Comment:
For the Hadronox case, I don't want them in the summon list specifically, as summon list has some specific behavior like assist on aggro etc.

For Gatewatcher case, that's a good point.